### DISCUSSION TITLE
relationship between tables with existing content

### DISCUSSION THREAD

**Discussion by Kurt-Aziri** [Budibase/budibase#1796] [Help]
I have two tables, which had a relationship. I deleted the relationship column in the 'many' table, since I intended to use a different field to create the relationship upon.
When I now want to define a new relationship column in the many table, this results in a new column in the parent table. I don't get this ? Shouldn't I just be able to selected the field upon which I want the relationship to be maintained ?

This is what I want to accomplish: I have a 'first name' and 'last name' field in the parent table. I created a formula column combining both fields (called 'name'). 
I would like to create a relationship from a 'many' table with this 'name' field in the parent table.
(As there is data in the parent and child tables, and I deleted the original relationship column, I expect to have to re-enter the relationship for each child record)

**Comment by mike12345567** [comment]
Hi @Kurt-Aziri, relationships in Budibase aren't really field based per say, when you create a relationship between two rows it is exactly that, between the two rows. It appears as if it is field level due to the tags which appear in the relationship column, this is based on the display name property of each table, for example you could set the display field to your `name` formula field and this would be displayed as the tag for the relationship.

This means that if you have a relationship between two tables you don't ever really need to delete it, you just need to change the display field of each table, then the tags shown for the relationship will change.

Below you can see how to set the display column, which will be used for the relationship.
![image](https://user-images.githubusercontent.com/4407001/122903307-2f061a80-d347-11eb-9ecc-14aaf8e6b502.png)

In the future it might be possible to override the display column and pick a specific column for the relationship to create its tags from, as not all relationships might be related to the default display column of a table.

**Reply by Kurt-Aziri** [reply]
Got it. It's not possible to set a formula field as a display column ?

**Reply by mike12345567** [reply]
It does appear thats the case, I'll look into that, likely an oversight on my part when I was building the formulas - will be a fix in the next release (this week).
